fil_fungible_token/src/token/mod.rs

```diff
@@ -278,8 +285,380 @@ where
         // - receiver hook should see the new balances...
 
         // if all succeeded, atomically commit the transaction
-        state.save(&self.bs)?;
-
+        self.save_state(&state)?;
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use cid::Cid;
+    use fvm_shared::econ::TokenAmount;
+    use num_traits::Zero;
+
+    use crate::blockstore::SharedMemoryBlockstore;
+
+    use super::Token;
+
+    #[test]
+    fn it_instantiates() {
+        let state_cid = Cid::default();
+        let mut token = Token::new(SharedMemoryBlockstore::new(), state_cid);
+        token.init_state().expect("state init failed");
+        token.load_state();
+    }
+
+    #[test]
+    fn it_persists_loads() {
+        let state_cid = Cid::default();
+        let bs = SharedMemoryBlockstore::new();
+        let mut token = Token::new(bs.clone(), state_cid);
+        token.init_state().unwrap();
+
+        // simulate actor calls mint method
+        let treasury = 1;
+        token.mint(treasury, TokenAmount::from(100)).unwrap();
+        let balance = token.balance_of(treasury).unwrap();
+        assert_eq!(balance, TokenAmount::from(100));
+
+        // here the actor would save the cid to it's state tree
+        // TODO: state_cid needs to be exported so that actors can keep track of it
+        let new_state_cid = token.state_cid;
```

Comment:
A pattern I'd expect here is `new_state_cid = token.save()` (even if, at the moment, save is no-op because it already saved)